sdk/personalizer/Azure.AI.Personalizer/src/Models/RankProcessor.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using Newtonsoft.Json;
+using Rl.Net;
+
+namespace Azure.AI.Personalizer
+{
+    /// <summary> The Rank Processor. </summary>
+    internal class RankProcessor
```

Comment:
I think we need a rankAzync method as well

Reply:
I agree with John. Async is only necessary if the method calls an asynchronous method internally or has a long running operation. Neither is present in this method so it shouldn't be required.